src/halmos/__main__.py

```diff
@@ -1357,30 +1366,19 @@ def parse_build_out(args: HalmosConfig) -> Dict:
                         contract_name,
                         sol_dirname,
                     )
-                contract_map[contract_name] = (json_out, contract_type, natspec)
 
-                try:
-                    bytecode = contract_map[contract_name][0]["bytecode"]["object"]
-                    contract_mapping_info = Mapper().get_contract_mapping_info_by_name(
-                        contract_name
-                    )
-
-                    if contract_mapping_info is None:
-                        Mapper().add_contract_mapping_info(
-                            contract_name=contract_name,
-                            bytecode=bytecode,
-                            nodes=[],
+                contract_map[contract_name] = (json_out, contract_type, natspec)
+                parse_symbols(args, contract_map, contract_name)
+
+            except KeyError as err:
+                if str(err) in ["ast", "metadata"]:
+                    # somewhat expected, e.g. for sol files containing only libraries
+                    if args.debug:
+                        debug(
+                            f"Skipped {json_filename} due to parsing failure: "
```

Comment:
this is the real change, the rest is a refactoring

Reply:
what's your intention when err is not ast nor metadata? generate warning, or silently ignore?

Reply:
another q: would it be possible to check if the ast / metadata missing error is indeed due to libraries?

Reply:
> another q: would it be possible to check if the ast / metadata missing error is indeed due to libraries?

that was the case every time I looked

Reply:
> what's your intention when err is not ast nor metadata? generate warning, or silently ignore?

oops, the intention is to generate a warning for other errors

Reply:
> > what's your intention when err is not ast nor metadata? generate warning, or silently ignore?
> 
> oops, the intention is to generate a warning for other errors

in that case, i don't think the current code works as intended.

Reply:
fixed in b135251